src/script/descriptor.cpp

```diff
@@ -1876,7 +1880,7 @@ std::unique_ptr<DescriptorImpl> InferScript(const CScript& script, ParseScriptCo
         CKeyID keyid(hash);
         CPubKey pubkey;
         if (provider.GetPubKey(keyid, pubkey)) {
-            if (auto pubkey_provider = InferPubkey(pubkey, ctx, provider)) {
+            if (auto pubkey_provider = InferPubkey(pubkey, ParseScriptContext::P2WPKH, provider)) {
```

Comment:
62ca75bd99b42f686ed1879481b856c882b40025: what's wrong with passing `ctx` here?

Reply:
The context for the pubkey inferring needs to be `P2WPKH`. `ctx` won't ever be that, instead it will either be `TOP` or `P2SH`, both of which allow uncompressed pubkeys, but P2WPKH does not. So the context needs to be updated in order to enforce that.